### PR TITLE
Remove unused variables.

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

**Comment by trivialfis** [comment]
cc @rongou

File: src/collective/communicator-inl.h [hunk 1]
```diff
@@ -252,7 +252,7 @@ inline AllgatherVResult<T> AllgatherV(std::vector<T> const &inputs,
 
   // Calculate input offsets (std::exclusive_scan).
   std::vector<std::size_t> offsets(all_sizes.size());
-  for (auto i = 1; i < offsets.size(); i++) {
+  for (std::size_t i = 1; i < offsets.size(); i++) {
```
Comment (rongou) [inline src/collective/communicator-inl.h:255]: What's the reason for not using `auto` here?

Comment (trivialfis) [reply src/collective/communicator-inl.h:255]: It's `int` by default, which generates a signed comparison warning when compared against `std::size_t` returned by `offsets.size()`.

Comment (rongou) [reply src/collective/communicator-inl.h:255]: Would it be simpler to append `u`?
```
for (auto i = 1u; i < offsets.size(); i++) {
```

Comment (trivialfis) [reply src/collective/communicator-inl.h:255]: I'm not sure which one is simpler. Matching types seems simple. ;-)

**Review by rongou — APPROVED** [review]

**Pull request merged**